test/YieldStrategyLeverTestBase.sol

```diff
@@ -0,0 +1,303 @@
+    bytes6 constant strategyIlkIdUSDC = 0x333300000000;
+    bytes6 constant strategyIlkIdETH = 0x333500000000;
+
+    bytes6 seriesId;
+    bytes6 strategyIlkId;
+    bytes12 vaultId;
+    address strategyTokenAddress;
+    uint256 baseAmount;
+    uint256 borrowAmount;
+    uint256 fyTokenToBuy;
+
+    constructor() {
+        giver = new Giver(cauldron);
+        // Orchestrate Giver
+        AccessControl cauldronAccessControl = AccessControl(
+            0xc88191F8cb8e6D4a668B047c1C8503432c3Ca867
+        );
+        vm.prank(timeLock);
+        cauldronAccessControl.grantRole(0x798a828b, address(giver));
+
+        vm.label(address(wethJoin), "weth Join");
+        vm.label(address(daiJoin), "dai Join");
+        vm.label(address(usdcJoin), "usdc Join");
+        vm.label(address(ladle.pools(seriesId)), "Pool");
+
+        vm.prank(daiWhale);
+        DAI.transfer(address(this), 100000e18);
+        vm.prank(usdcWhale);
+        USDC.transfer(address(this), 100000e6);
+        vm.prank(wethWhale);
+        WETH.transfer(address(this), 100000e18);
+
+        vm.prank(timeLock);
+        wethJoin.setFlashFeeFactor(1);
+        vm.prank(timeLock);
+        usdcJoin.setFlashFeeFactor(1);
+        vm.prank(timeLock);
+        daiJoin.setFlashFeeFactor(1);
+    }
+
+    function setUp() public virtual {
+        IPool pool = IPool(ladle.pools(seriesId));
+        FYToken fyToken = FYToken(address(pool.fyToken()));
+        vm.label(address(fyToken), fyToken.symbol());
+        vm.label(address(pool.base()), pool.baseToken().symbol());
+        vm.label(address(pool), "Pool");
+        vm.prank(timeLock);
+        fyToken.setFlashFeeFactor(1);
+
+        lever = new YieldStrategyLever(giver);
+
+        DAI.approve(address(lever), type(uint256).max);
+        USDC.approve(address(lever), type(uint256).max);
+        WETH.approve(address(lever), type(uint256).max);
+
+        AccessControl giverAccessControl = AccessControl(address(giver));
+        giverAccessControl.grantRole(0xe4fd9dc5, timeLock);
+        giverAccessControl.grantRole(0x35775afb, address(lever));
+    }
+
+    /// @notice Create a vault.
+    function leverUp() public returns (bytes12) {
+        vaultId = lever.invest(
+            YieldStrategyLever.Operation.BORROW,
+            seriesId,
+            strategyIlkId, // ilkId edai
+            baseAmount,
+            borrowAmount,
+            fyTokenToBuy,
+            0 //minCollateral,
+        );
+        return vaultId;
+    }
+}
+
+abstract contract Tests is ZeroState {
+    function setUp() public virtual override {
+        super.setUp();
+        vaultId = leverUp();
+    }
+
+    /// @notice Test if vault is created correctly
+    function testVault() public {
+        vaultId = leverUp();
+        DataTypes.Vault memory vault = cauldron.vaults(vaultId);
+        assertEq(vault.owner, address(this));
+
+        assertEq(cauldron.balances(vaultId).art, borrowAmount);
```

Comment:
I would also check `cauldron.balances(vaultId).ink` and `strategy.balanceOf(vault.owner)`